test/lkm_samples/cwe_252.c

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: GPL-2.0
+
+#include <linux/module.h>
+#include <linux/printk.h>
+#include <linux/slab.h>
+#include <linux/string.h>
+#include <linux/uaccess.h>
+#include <asm/rwonce.h>
+
+void cwe252_intra_h_lost(void* uptr)
+{
+  char buf[10];
+  long *ret_store = kmalloc(8, GFP_KERNEL);
+
+  WRITE_ONCE(*ret_store, strncpy_from_user(buf, uptr, sizeof(buf))); // CWE_WARNING, 1
+
+  pr_info("Call some func\n");
+
+  WRITE_ONCE(*ret_store, 0); // CWE_WARNING, 2, reason=empty_state
+
+  pr_info("buf: %s\n", buf);
+}
+
+static int __init test_init(void)
+{
+    pr_info("Hello, World\n");
+
+    return 0;
+}
+
+static void __exit test_exit(void)
+{
+    pr_info("Goodbye, World\n");
+}
+
+MODULE_LICENSE("GPL v2");
```

Comment:
We can keep it. I just wondered whether we now have to mention in the `Readme.md` that parts of the repository are GPL instead of LGPL. But it is only test code, I guess we can just quietly ignore what we tell the Linux kernel here and treat it as licensed like the rest of the repository.